src/test/java/org/folio/rest/api/ItemStorageTest.java

```diff
@@ -269,6 +269,165 @@ public void canCreateAnItemWithoutProvidingID()
+  }
+
+  @Test
+  public void canCreateAnItemWithItemLevelCallNumberAsEffectiveCallNumber()
+    throws MalformedURLException, InterruptedException,
+    ExecutionException, TimeoutException {
+
+    UUID holdingsRecordId = createInstanceAndHoldingWithCallNumber(mainLibraryLocationId);
+
+    JsonObject itemToCreate = nod(null, holdingsRecordId);
+
+    itemToCreate.put("tags", new JsonObject().put("tagList", new JsonArray().add(TAG_VALUE)));
+
+    itemToCreate.put("itemLevelCallNumber", "testItemCallNumber");
+
+    CompletableFuture<Response> createCompleted = new CompletableFuture<>();
+
+    client.post(itemsStorageUrl(""), itemToCreate, StorageTestSuite.TENANT_ID,
+      ResponseHandler.json(createCompleted));
+
+    Response postResponse = createCompleted.get(5, TimeUnit.SECONDS);
+
+    assertThat(postResponse.getStatusCode(), is(HttpURLConnection.HTTP_CREATED));
+
+    JsonObject itemFromPost = postResponse.getJson();
+
+    String newId = itemFromPost.getString("id");
+
+    assertThat(newId, is(notNullValue()));
+
+    Response getResponse = getById(UUID.fromString(newId));
+
+    assertThat(getResponse.getStatusCode(), is(HttpURLConnection.HTTP_OK));
+
+    JsonObject itemFromGet = getResponse.getJson();
+
+    assertThat(itemFromGet.getString("id"), is(newId));
+    assertThat(itemFromGet.getString("holdingsRecordId"), is(holdingsRecordId.toString()));
+    assertThat(itemFromGet.getString("barcode"), is("565578437802"));
+    assertThat(itemFromGet.getJsonObject("status").getString("name"),
+      is("Available"));
+    assertThat(itemFromGet.getString("materialTypeId"),
+      is(journalMaterialTypeID));
+    assertThat(itemFromGet.getString("permanentLoanTypeId"),
+      is(canCirculateLoanTypeID));
+    assertThat(itemFromGet.getString("temporaryLocationId"),
+      is(annexLibraryLocationId.toString()));
+
+    List<String> tags = itemFromGet.getJsonObject("tags").getJsonArray("tagList").getList();
+
+    assertThat(tags.size(), is(1));
+    assertThat(tags, hasItem(TAG_VALUE));
+    assertThat(itemFromGet.getString("effectiveCallNumber"), is("testItemCallNumber"));
+  }
+
+  @Test
+  public void canUpdateAnItemWithHoldingCallNumberAsEffectiveCallNumber()
+    throws MalformedURLException, InterruptedException,
+    ExecutionException, TimeoutException {
+
+    UUID holdingsRecordId = createInstanceAndHoldingWithCallNumber(mainLibraryLocationId);
+    JsonObject itemToCreate = new JsonObject();
+    String itemId = UUID.randomUUID().toString();
+    itemToCreate.put("id", itemId);
+    itemToCreate.put("holdingsRecordId", holdingsRecordId.toString());
+    itemToCreate.put("permanentLoanTypeId", canCirculateLoanTypeID);
+    itemToCreate.put("materialTypeId", bookMaterialTypeID);
+    createItem(itemToCreate);
+
+    CompletableFuture<Response> completed = new CompletableFuture<>();
+    client.put(itemsStorageUrl("/" + itemId), itemToCreate, StorageTestSuite.TENANT_ID,
```

Comment:
This appears to update the item straight after it is created with no changes to either the item or the holding.

Can you help me understand the scenario this is trying to demonstrate?

Reply:
This is a test to show that when we create an item and that item does not have its own itemLevelCallNumber, that the holding's callNumber will be set as the item's effectiveCallNumber during item creation.

Line 381 creates a holding with callNumber 'testCallNumber'.
Line 385 associates the holding with the soon to be created item.
Line 400 verifies that 'testCallNumber' has been set as the item's effectiveCallNumber.

Reply:
The name of this test, canUpdateAnItemWithHoldingCallNumberAsEffectiveCallNumber, suggests it has something to do with the derivation of the effective call number after the item is updated.

Your description suggests it is purely about item creation.

These seem like different situations to me, have I misunderstood?